assembly.xml

```diff
@@ -0,0 +1,111 @@
+                     a source-release archive that comes from a pretty clean directory
+                     structure.
+
+                     HOWEVER, if the release plugin is configured to run extra goals
+                     or generate a project website, it's definitely possible that some
+                     of these files will be present. So, it's safer to exclude them.
+                -->
+
+                <!-- IDEs -->
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?maven-eclipse\.xml]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.project]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.classpath]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?[^/]*\.iws]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.idea(/.*)?]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?out(/.*)?]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?[^/]*\.ipr]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?[^/]*\.iml]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.settings(/.*)?]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.externalToolBuilders(/.*)?]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.deployables(/.*)?]</exclude>
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?\.wtpmodules(/.*)?]</exclude>
+
+                <!-- misc -->
+                <exclude>%regex[(?!((?!${project.build.directory}/)[^/]+/)*src/)(.*/)?cobertura\.ser]</exclude>
+
+                <!-- files generated by the Gradle build process. -->
+                <exclude>**/.gradle</exclude>
+                <exclude>**/.gogradle</exclude>
+                <exclude>**/build</exclude>
+                <exclude>**/vendor</exclude>
+                <exclude>**/.gradletasknamecache</exclude>
+
+                <!-- files generated by the Maven build process. -->
+                <exclude>**/bin</exclude>
+                <exclude>**/dependency-reduced-pom.xml</exclude>
+
+                <!-- file generated for archetypes -->
+                <exclude>sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/src/</exclude>
+                <exclude>sdks/java/maven-archetypes/examples-java8/src/main/resources/archetype-resources/src/</exclude>
+
+                <!-- files generated by the Python build process. -->
+                <exclude>**/*.py[cod]</exclude>
+                <exclude>**/*.egg-info/</exclude>
+                <exclude>**/.eggs/</exclude>
```

Comment:
This one seems not to be working, it inlcudes the python.eggs subdir with its children *.egg directories

Reply:
OK, let me try.

Reply:
Which file did you check ? `apache-beam-2.3.0-SNAPSHOT-source-release.zip` or `apache-beam-2.3.0-SNAPSHOT-src.zip` in the target folder ?

Reply:
The `src` is fine, the `source` (coming from the Apache POM) is not. Let me exclude this one.